### PR TITLE
grub2: apply security patches

### PR DESCRIPTION
**Pull request by LeSuisse**
###### Description of changes

Apply security patches for CVE-2021-3981, CVE-2021-3695, CVE-2021-3696, CVE-2021-3697, CVE-2022-28733, CVE-2022-28734, CVE-2022-28735, CVE-2022-28736, CVE-2022-28737, CVE-2022-2601 and CVE-2022-3775.
    
https://lists.gnu.org/archive/html/grub-devel/2022-06/msg00035.html
https://lists.gnu.org/archive/html/grub-devel/2022-11/msg00059.html

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>grub2 (grub2_light)</li>
    <li>grub2_efi</li>
    <li>grub2_pvgrub_image</li>
    <li>grub2_xen</li>
    <li>os-prober</li>
    <li>timeshift</li>
    <li>woeusb</li>
    <li>woeusb-ng</li>
  </ul>
</details>

### CODE REVIEW

**Comment by LeSuisse** [comment]
`grub2.passthru.tests.nixos-install-grub1` and `grub2.passthru.tests.nixos-install-grub-uefi-spec` are failing but they also seems to fail on `master` for me.

**Comment by RaitoBezarius** [comment]
Can look into the test failures.

**Comment by Artturin** [comment]
it appears that its running out of memory
```
vm-test-run-installer-simpleUefiGrubSpecialisation> machine # building '/nix/store/i6l9z9rvjkflvbdapn8shfby7q709603-nixos-manual-html.drv'...
vm-test-run-installer-simpleUefiGrubSpecialisation> machine # [   81.680032] xsltproc invoked oom-killer: gfp_mask=0x1100dca(GFP_HIGHUSER_MOVABLE|__GFP_ZERO), order=0, oom_score_adj=0
```

i've pushed a fix

File: nixos/tests/installer.nix [hunk 1]
```diff
@@ -291,7 +291,7 @@ let
           # builds stuff in the VM, needs more juice
           virtualisation.diskSize = 8 * 1024;
           virtualisation.cores = 8;
-          virtualisation.memorySize = 1536;
+          virtualisation.memorySize = 2047;
```
Comment (SuperSandro2000) [inline nixos/tests/installer.nix:294]: Is there a reason this is just one MB shy of 2048?

Comment (Artturin) [reply nixos/tests/installer.nix:294]: https://github.com/NixOS/nixpkgs/commit/aa3b13549fbd4d5da27bdf07b6cb66555e837b8e
https://hydra.nixos.org/build/24983009
on i686-linux

`machine# qemu: at most 2047 MB RAM can be simulated`

**Comment by LeSuisse** [comment]
Good catch, thanks.

`grub2.passthru.tests.nixos-install-grub-uefi-spec` still failing on my end, it works if I disable the build of the NixOS documentation:
```diff
diff --git a/nixos/tests/installer.nix b/nixos/tests/installer.nix
index 300f6602789..258411305b1 100644
--- a/nixos/tests/installer.nix
+++ b/nixos/tests/installer.nix
@@ -452,6 +452,7 @@ let
     extraConfig = ''
       environment.systemPackages = [ pkgs.grub2 ];
       boot.loader.grub.configurationName = "Home";
+      documentation.nixos.enable = false;
       specialisation.work.configuration = {
         boot.loader.grub.configurationName = lib.mkForce "Work";
 
```

Is this good enough for what we are trying to test?

For `grub2.passthru.tests.nixos-install-grub1` the error seems to come from soemthing else:
```
machine # updating GRUB 1 menu...
machine # installing the GRUB 1 boot loader on /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive1...
machine # [  111.029383] process 'store/m14gcv7hr5jzwm49dyyifa54vx3jamm5-grub-0.97-73/bin/grub' started with executable stack
machine # Probing devices to guess BIOS drives. This may take a long time.
machine # [  111.102785] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  111.103646] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # /nix/store/m14gcv7hr5jzwm49dyyifa54vx3jamm5-grub-0.97-73/sbin/grub-install: line 468: grub-probe: command not found
machine # Unknown partition table signature
machine # [  113.606769] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  113.607634] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  119.718740] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  119.719860] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  131.832669] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  131.833542] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  137.933613] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  137.934336] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  144.013657] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  144.014462] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  150.188611] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  150.189445] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  156.288559] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  156.289406] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  162.545476] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  162.546327] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  168.708465] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  168.709326] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  174.967421] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  174.968249] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  181.055388] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  181.056256] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  187.229321] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  187.230414] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  193.377277] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  193.378123] floppy: error 10 while reading block 0
machine # 
machine # 
machine #     GNU GRUB  version 0.97  (640K lower / 3072K upper memory)
machine # 
machine #        [ Minimal BASH-like line editing is supported.   For
machine #          the   first   word,  TAB  lists  possible  command
machine #          completions.  Anywhere else TAB lists the possible
machine #          completions of a device/filename. ]
machine # grub> root (hd0,1)
machine #  Filesystem type is ext2fs, partition type 0x83
machine # grub> setup  --stage2=/tmp/PXzqqrtILy/boot/grub/stage2 --prefix=/boot/grub (hd0)
machine #  Checking if "/boot/grub/stage1" exists... yes
machine #  Checking if "/boot/grub/stage2" exists... yes
machine #  Checking if "/boot/grub/e2fs_stage1_5" exists... yes
machine #  Running "embed /boot/grub/e2fs_stage1_5 (hd0)"... failed (this is not fatal)
machine #  Running "embed /boot/grub/e2fs_stage1_5 (hd0,1)"... failed (this is not fatal)
machine #  Running "install --stage2=/tmp/PXzqqrtILy/boot/grub/stage2 /boot/grub/stage1 (hd0) /boot/grub/stage2 p /boot/grub/menu.lst "... failed
machine # 
machine # Error 6: Mismatched or corrupt version of stage1/stage2
machine # grub> quit
machine # /nix/store/bgdk99syw9bfg45irav36na1jiyrkqfz-install-grub.pl: installation of GRUB on /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive1 failed: No such file or directory
```

The `line 468: grub-probe: command not found` does not seem expected.

**Comment by LeSuisse** [comment]
Good catch, thanks.

`grub2.passthru.tests.nixos-install-grub-uefi-spec` still failing on my end, it works
if I disable the build of the NixOS documentation:
```diff
diff --git a/nixos/tests/installer.nix b/nixos/tests/installer.nix
index 300f6602789..258411305b1 100644
--- a/nixos/tests/installer.nix
+++ b/nixos/tests/installer.nix
@@ -452,6 +452,7 @@ let
     extraConfig = ''
       environment.systemPackages = [ pkgs.grub2 ];
       boot.loader.grub.configurationName = "Home";
+      documentation.nixos.enable = false;
       specialisation.work.configuration = {
         boot.loader.grub.configurationName = lib.mkForce "Work";
 
```

Is this good enough for what we are trying to test?

For `grub2.passthru.tests.nixos-install-grub1` the error seems to come from soemthing else:
```
machine # updating GRUB 1 menu...
machine # installing the GRUB 1 boot loader on /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive1...
machine # [  111.029383] process 'store/m14gcv7hr5jzwm49dyyifa54vx3jamm5-grub-0.97-73/bin/grub' started with executable stack
machine # Probing devices to guess BIOS drives. This may take a long time.
machine # [  111.102785] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  111.103646] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # /nix/store/m14gcv7hr5jzwm49dyyifa54vx3jamm5-grub-0.97-73/sbin/grub-install: line 468: grub-probe: command not found
machine # Unknown partition table signature
machine # [  113.606769] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  113.607634] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  119.718740] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  119.719860] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  131.832669] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  131.833542] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  137.933613] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  137.934336] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  144.013657] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  144.014462] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  150.188611] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  150.189445] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  156.288559] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  156.289406] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  162.545476] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  162.546327] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  168.708465] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  168.709326] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  174.967421] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  174.968249] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  181.055388] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  181.056256] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  187.229321] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  187.230414] floppy: error 10 while reading block 0
machine # Unknown partition table signature
machine # [  193.377277] blk_update_request: I/O error, dev fd0, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
machine # [  193.378123] floppy: error 10 while reading block 0
machine # 
machine # 
machine #     GNU GRUB  version 0.97  (640K lower / 3072K upper memory)
machine # 
machine #        [ Minimal BASH-like line editing is supported.   For
machine #          the   first   word,  TAB  lists  possible  command
machine #          completions.  Anywhere else TAB lists the possible
machine #          completions of a device/filename. ]
machine # grub> root (hd0,1)
machine #  Filesystem type is ext2fs, partition type 0x83
machine # grub> setup  --stage2=/tmp/PXzqqrtILy/boot/grub/stage2 --prefix=/boot/grub (hd0)
machine #  Checking if "/boot/grub/stage1" exists... yes
machine #  Checking if "/boot/grub/stage2" exists... yes
machine #  Checking if "/boot/grub/e2fs_stage1_5" exists... yes
machine #  Running "embed /boot/grub/e2fs_stage1_5 (hd0)"... failed (this is not fatal)
machine #  Running "embed /boot/grub/e2fs_stage1_5 (hd0,1)"... failed (this is not fatal)
machine #  Running "install --stage2=/tmp/PXzqqrtILy/boot/grub/stage2 /boot/grub/stage1 (hd0) /boot/grub/stage2 p /boot/grub/menu.lst "... failed
machine # 
machine # Error 6: Mismatched or corrupt version of stage1/stage2
machine # grub> quit
machine # /nix/store/bgdk99syw9bfg45irav36na1jiyrkqfz-install-grub.pl: installation of GRUB on /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive1 failed: No such file or directory
```

The `line 468: grub-probe: command not found` does not seem expected.

File: pkgs/tools/misc/grub/2.0x.nix [hunk 2]
```diff
@@ -78,6 +78,68 @@ stdenv.mkDerivation rec {
...
+      name = "CVE-2022-2601.CVE-2022-3775.6.patch";
+      url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=b9396daf1c2e3cdc0a1e69b056852e0769fb24de";
+      sha256 = "sha256-K7XNneDZjLpZh/C908+5uYsB/0oIdgQqmk0yJrdQLG4=";
+    })
+    (fetchpatch {
+      name = "CVE-2022-2601.CVE-2022-3775.7.patch";
+      url = "https://git.savannah.gnu.org/cgit/grub.git/patch/?id=1d2015598cc7a9fca4b39186273e3519a88e80c7";
+      sha256 = "sha256-s4pZtszH4b/0u85rpzVapZmNQdYEq/wW06SQ3PW/1aU=";
+    })
+    # Patch 8 should be https://git.savannah.gnu.org/gitweb/?p=grub.git;a=commit;h=93a786a00163e50c29f0394df198518617e1c9a5
```
Comment (GovanifY) [inline pkgs/tools/misc/grub/2.0x.nix:116]: Looking at the patch, is this not something that could apply on grub2_efi? Why would we have removed entirely the shim logic from our source tree...?
I'm probably missing some context here.

**Review by GovanifY — COMMENTED** [review]
The patch seems good, and tbh I don't think GRUB1 tests failing is a big deal as 
1. You did not touch grub1 at all, those are purely grub2 upstream backports
2. GRUB1 is dead.

I'm still curious about the sb.c patch specifically but otherwise LGTM from a cursory look.

File: pkgs/tools/misc/grub/2.0x.nix (hunk 2 above)
Comment (LeSuisse) [reply pkgs/tools/misc/grub/2.0x.nix:116]: In fact we are missing the patch for CVE-2022-28735 which explains why it does not apply.

**Comment by LeSuisse** [comment]
Moving back this PR to draft.
It appears we are also missing patches for CVE-2022-28736, CVE-2022-28734, CVE-2022-28733, CVE-2021-3697, CVE-2021-3696 and CVE-2021-3695. I will try to deal with it later today or tomorrow.

**Comment by SuperSandro2000** [comment]
Wouldn't mind if grub did another release.

**Comment by LeSuisse** [comment]
Ready for review again. Hopefully all patches are applied.

All `grub2` tests are successful on my side.

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 208614` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>grub2 (grub2_full ,grub2_light)</li>
    <li>grub2_efi</li>
    <li>grub2_pvgrub_image</li>
    <li>grub2_xen</li>
    <li>os-prober</li>
    <li>tests.testers.nixosTest-example</li>
    <li>timeshift</li>
    <li>woeusb</li>
    <li>woeusb-ng</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 208614` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>woeusb</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>grub2_pvgrub_image</li>
    <li>grub2_xen</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>grub2 (grub2_full ,grub2_light)</li>
    <li>grub2_efi</li>
    <li>timeshift</li>
    <li>woeusb-ng</li>
  </ul>
</details>

**Comment by LeSuisse** [comment]
It looks like the failures on aarch64 are not new:
https://hydra.nixos.org/build/202124533
https://hydra.nixos.org/build/202185637

```
configure: error: platform "xen" is not supported for target CPU "arm64"
```

**Comment by Artturin** [comment]
> Good catch, thanks.
> 
> `grub2.passthru.tests.nixos-install-grub-uefi-spec` still failing on my end, it works if I disable the build of the NixOS documentation:
> 
> ```diff
> diff --git a/nixos/tests/installer.nix b/nixos/tests/installer.nix
> index 300f6602789..258411305b1 100644
> --- a/nixos/tests/installer.nix
> +++ b/nixos/tests/installer.nix
> @@ -452,6 +452,7 @@ let
>      extraConfig = ''
>        environment.systemPackages = [ pkgs.grub2 ];
>        boot.loader.grub.configurationName = "Home";
> +      documentation.nixos.enable = false;
>        specialisation.work.configuration = {
>          boot.loader.grub.configurationName = lib.mkForce "Work";
>  
> ```
> 
> Is this good enough for what we are trying to test?
> 


did you forget to pull the fix because when you force pushed you removed my fix

**Comment by LeSuisse** [comment]
> > Good catch, thanks.
> > `grub2.passthru.tests.nixos-install-grub-uefi-spec` still failing on my end, it works if I disable the build of the NixOS documentation:
> > ```diff
> > diff --git a/nixos/tests/installer.nix b/nixos/tests/installer.nix
> > index 300f6602789..258411305b1 100644
> > --- a/nixos/tests/installer.nix
> > +++ b/nixos/tests/installer.nix
> > @@ -452,6 +452,7 @@ let
> >      extraConfig = ''
> >        environment.systemPackages = [ pkgs.grub2 ];
> >        boot.loader.grub.configurationName = "Home";
> > +      documentation.nixos.enable = false;
> >        specialisation.work.configuration = {
> >          boot.loader.grub.configurationName = lib.mkForce "Work";
> >  
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Is this good enough for what we are trying to test?
> 
> did you forget to pull the fix because when you force pushed you removed my fix

Nope I tested with your fix and I was still OOM. With `documentation.nixos.enable = false;` I did not encountered the issue and the test was succeeding on the OfBorg run.

**Comment by Artturin** [comment]
```
error: builder for '/nix/store/mpxfkmj5jdw4n3ny7zq4v58fxdc1rds0-CVE-2021-3695.CVE-2021-3696.CVE-2021-3697.CVE-2022-28733.CVE-2022-28734.CVE-2022-28735.CVE-2022-28736.CVE-2022-28737.29.patch.drv' failed with exit code 1;
       last 10 log lines:
       > Warning: Problem : HTTP error. Will retry in 1 seconds. 3 retries left.
       >   0   234    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 503
       > Warning: Problem : HTTP error. Will retry in 2 seconds. 2 retries left.
       >   0   234    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 503
       > Warning: Problem : HTTP error. Will retry in 4 seconds. 1 retries left.
       >   0   234    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 503
       > error: cannot download CVE-2021-3695.CVE-2021-3696.CVE-2021-3697.CVE-2022-28733.CVE-2022-28734.CVE-2022-28735.CVE-2022-28736.CVE-2022-28737.29.patch from any mirror
       For full logs, run 'nix log /nix/store/mpxfkmj5jdw4n3ny7zq4v58fxdc1rds0-CVE-2021-3695.CVE-2021-3696.CVE-2021-3697.CVE-2022-28733.CVE-2022-28734.CVE-2022-28735.CVE-2022-28736.CVE-2022-28737.29.patch.drv'.
```

weird ofborg is failing with lots of these

**Comment by cole-h** [comment]
@ofborg build grub2 grub2.passthru.tests

**Comment by cole-h** [comment]
FYI: the patch download failures were some kind of limiting (rate limiting / concurrent connection limiting? :shrug: ) (thanks Lily Foster on Matrix for the idea). After making ofborg rebuild it a few times, we finally got enough to continue. The only failure on x86 now is the grub1 test (which, as mentioned above, is also failing on master).

**Comment by RaitoBezarius** [comment]
I think this is ready to be merged.

**Comment by vcunat** [comment]
Uh, this caused a problem:
https://github.com/NixOS/nixpkgs/pull/211372

**Pull request merged**